strings/string_is_valid_number.py

```diff
@@ -0,0 +1,134 @@
+"""
+Solution By: Reniz Shah
+Topic: Deterministic Finite Automaton (DFA)
+Given a string s, return whether s is a valid number or not
+Input: s = -90E3
+Output: True
+Leetcode link: https://leetcode.com/problems/valid-number/description/
+"""
+
+
+def classify_char(char):
+    """
+    Classifies a character into one of the following categories:
+
+    - 'numeric': if the character is a digit (0-9)
+    - 'ign': if the character is a plus sign (+) or a minus sign (-)
+    - 'exponent': if the character is an 'e' or 'E' (used in exponential notation)
+    - 'decimal': if the character is a decimal point (.)
+    - None: if the character does not fit into any of the above categories
+
+    Parameters:
+    char (str): The character to be classified
+
+    Returns:
+    str: The classification of the character
+
+    >>> classify_char('2')
+    'numeric'
+    >>> classify_char('-')
+    'sign'
+    >>> classify_char('e')
+    'exponent'
+    >>> classify_char('.')
+    'decimal'
+    >>> classify_char('r')
```

Comment:
```suggestion
    >>> classify_char('r') = None
    True
```
Please add tests for `""`, `"01"`, and `0`

Reply:
For `"01"` (and all other strings that aren't single chars), this function should raise an exception.